soc/xtensa/esp32/dts_fixup.h

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2020 Mohamed ElShahawi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* SoC level DTS fixup file */
+
+#ifdef CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+#undef CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC
+#endif
+/* Timers Run @ APB Clock Speed 80MHZ */
+#define CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC MHZ(80) /* FIXME: Add #if for PLL/XTAL Sources */
```

Comment:
This isn't really kosher.  Don't redefine kconfig values to clobber them with devicetree settings.  If you have to get the frequency from dts we probably need another framework.

Also note that the general promise of the hardware cycle counter API is that the rate is invariant.  If that's not true anymore (I can't tell) then you probably want to investigate the sys_clock_hw_cycles_per_sec() wrapper API instead.

Reply:
I agree this is not the right way, it is here to show the issue I'm facing.
I Will check `sys_clock_hw_cycles_per_sec` and do some timing tests.
The problem is the Clock speed of timers is different when PLL is Enabled vs Disabled.

Reply:
Now `sys_clock_hw_cycles_per_sec` checked using `BUILD_ASSERT` in the driver.
It must always equal to CPU CLK.